lib/util/trip/ride_card.dart

```diff
@@ -1,84 +1,195 @@
+          ))
+        : Padding(
+            padding: const EdgeInsets.fromLTRB(16, 8, 16, 16),
+            child: ProfileWidget(_driver!),
+          );
+  }
+
+  @override
+  Widget buildBottomRight() {
+    return !_fullyLoaded
+        ? const Center(
+            child: SizedBox(
+            height: 44,
+            width: 52,
+          ))
+        : Padding(
+            padding: const EdgeInsets.fromLTRB(16, 8, 16, 16),
+            child: CustomRatingBarIndicator(
+                rating: AggregateReview.fromReviews(_ride!.drive!.driver!.reviewsReceived!).rating,
+                size: CustomRatingBarSize.medium),
+          );
+  }
+
+  @override
+  Widget buildRightSide() {
+    if (!_fullyLoaded) {
+      return const Center(
+          child: SizedBox(
+        height: 24,
+        width: 24,
+      ));
+    } else {
+      List<ProfileFeature> profileFeatures = _driver!.profileFeatures!;
+      List<Icon> featureicons = <Icon>[];
+      for (int i = 0; i < min(profileFeatures.length, 3); i++) {
+        featureicons.add(profileFeatures[i].feature.getIcon(context));
+      }
+      return Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: featureicons,
+      );
+    }
+  }
+
+  Color pickBannerColor() {
+    switch (_ride!.status) {
+      case RideStatus.preview:
+        return Theme.of(context).cardColor;
+      case RideStatus.pending:
+        return Theme.of(context).disabledColor;
+      case RideStatus.approved:
+        return Theme.of(context).own().success;
+      case RideStatus.rejected:
+        return Theme.of(context).colorScheme.error;
+      case RideStatus.cancelledByDriver:
+        return Theme.of(context).colorScheme.error;
+      case RideStatus.cancelledByRider:
+        return Theme.of(context).colorScheme.error;
+      case RideStatus.withdrawnByRider:
+        return Theme.of(context).cardColor;
+    }
+  }
+
+  BoxDecoration pickDecoration() {
+    if (_ride!.status.isCancelled() || _ride!.status == RideStatus.rejected) {
+      return BoxDecoration(
+        color: Colors.grey,
+        borderRadius: cardBorder,
+        backgroundBlendMode: BlendMode.screen,
+      );
+    } else {
+      return BoxDecoration(borderRadius: cardBorder);
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     return Card(
-      child: Semantics(
-        button: true,
-        tooltip: S.of(context).openDetails,
-        child: InkWell(
-          onTap: () => Navigator.of(context).push(
-            MaterialPageRoute(
-              builder: (context) => RideDetailPage.fromRide(trip),
+      color: pickBannerColor(),
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(10),
+      ),
+      child: Wrap(
+        children: [
+          Container(
+            foregroundDecoration: pickDecoration(),
+            width: MediaQuery.of(context).size.width,
+            decoration: BoxDecoration(
+              color: Theme.of(context).cardColor,
```

Comment:
this does not seem to work, the whole card is white

Reply:
Do you mean when the card is disabled or the normal cards?

Reply:
the normal cards, see here:

<img width="335" alt="image" src="https://user-images.githubusercontent.com/63241108/213172329-55ec4b90-dd03-41a2-a399-7b9ac37cfe21.png">

Reply:
Card color is fine, but it should be applied to the whole card.